plan/runner.go

```diff
@@ -124,6 +126,15 @@ func (r *Runner) shouldRun(p cue.Path) bool {
 	return ok
 }
 
+func taskLog(tp string, log *zerolog.Logger, t task.Task, fn func(lg zerolog.Logger)) {
+	fn(log.With().Str("task", tp).Logger())
+	// setup logger here
+	_, isDockerfileTask := t.(*task.DockerfileTask)
+	if isDockerfileTask {
+		fn(log.With().Str("task", "system").Logger())
```

Comment:
I don't get this part -- if I understand correctly, for Dockerfile tasks, we're logging things twice -- once with `task = ""` and once again with `task = "system"`, then on the tty/plain writers we rewrite `"" -> "system"`.

I don't understand why this is needed?

Reply:
At this point we're not logging with `task = ""` since we're inside the cueflow taskFunc and we have the correct Dagger task names. 

We're logging twice because otherwise, the task action name won't be printed to the tty. So, for `dockerfile` actions, at this point we need to log twice so both the dockerfile action and the `system` logger get recorded by the tty output.

Reply:
> Sorry, I'm slow today, still not following :)

np haha! It's likely that I'm missing something here as well. 

> The second log call doesn't contain the task name, right? I don't see why it would change what's printed.

Correct, but we need the second log so the `system` group is state updated and handled by the tty output here: https://github.com/dagger/dagger/blob/main/cmd/dagger/logger/tty.go#L262

> I don't understand the goal of the system entry since we can't rely on computing/completed etc (as you can have a bunch of Dockerfile running in parallel)

You're seeing this in the `plain` output, but the `tty` output needs this state transition so it can correctly display if something failed or not. In the case you have multiple DAG nodes with Dockerfile actions, `system` will have the state of the last execution.

Reply:
Got it!

I think there are some edge cases as the "last" execution is fuzzy: you can have N Dockerfile running in parallel, e.g. the state changes will not be linear (#1 computing / #2 computing / #2 completed / #1 completed), so tty will "flip" between computing/completed for the same docker build.

However, I don't know if this is really a big problem in practice and since this PR is a giant improvement already over what we have, I vote we merge it as is and see how it goes :)

Reply:
Let's not merge this yet. I want to explore something to see if it can be further improved before merging